src/libs/getAgreementArticles.js

```diff
@@ -0,0 +1,35 @@
+// @ts-check
+
+const unistUtilFlatFilter = require("unist-util-flat-filter");
+
+const getAgreement = require("./getAgreement");
+const getArticle = require("./getArticle");
+
+/**
+ * @param {number | string} agreementIdOrIdcc
+ *
+ * @returns {KaliData.AgreementArticleWithSections[]}
+ */
+function getAgreementArticles(agreementIdOrIdcc) {
+  const agreement = getAgreement(agreementIdOrIdcc);
+
+  const articles =
+    /** @type {{ children: KaliData.AgreementArticle[], type: "root" }} */
+    (/** @type {*} */ (unistUtilFlatFilter(agreement, { type: "article" })));
+  const articleIds = articles.children.map(({ data: { id } }) => id);
+
+  const articlesWithSections = articleIds.reduce((prevArticlesWithSections, articleId) => {
```

Comment:
not sure to understand why it's named `articlesWithSections` since you only return node with `type === article`

Reply:
The next line allow us to retrieve an article with its parent sections: `const articleWithSections = getArticle(articleId);`

`getArticle` could in fact be named `getArticleWithSections`: https://github.com/SocialGouv/kali-data/blob/ivang/feat-add-set-of-helpers/src/libs/getArticle.js#L45